doc/source/selections.rst

```diff
@@ -0,0 +1,440 @@
+
+Geometric
+---------
+
+around *distance*  *selection*
+    selects all atoms a certain cutoff away from another selection,
+    e.g. ``around 3.5 protein`` selects all atoms not belonging to protein
+    that are within 3.5 Angstroms from the protein
+
+sphlayer *innerRadius* *externalRadius* *selection*
+    selects all atoms within a spherical layer centered in the center of
+    geometry (COG) of a given selection, e.g., ``sphlayer 2.4 6.0 ( protein
+    and ( resid 130 or resid 80 ) )`` selects the center of geometry of
+    protein, resid 130, resid 80 and creates a spherical layer of inner
+    radius 2.4 and external radius 6.0 around the COG.
+
+sphzone *externalRadius* *selection*
+    selects all atoms within a spherical zone centered in the center of
+    geometry (COG) of a given selection, e.g. ``sphzone 6.0 ( protein and (
+    resid 130 or resid 80 ) )`` selects the center of geometry of protein,
+    resid 130, resid 80 and creates a sphere of radius 6.0 around the COG.
+
+cylayer *innerRadius* *externalRadius* *zMax* *zMin* *selection*
+    selects all atoms within a cylindric layer centered in the center of
+    geometry (COG) of a given selection, e.g. ``cylayer 5 10 10 -8
+    protein`` selects the center of geometry of protein, and creates a
+    cylindrical layer of inner radius 5, external radius 10 centered on the
+    COG. In z, the cylinder extends from 10 above the COG to 8
+    below. Positive values for *zMin*, or negative ones for *zMax*, are
+    allowed.
+
+cyzone *externalRadius* *zMax* *zMin* *selection*
+    selects all atoms within a cylindric zone centered in the center of
+    geometry (COG) of a given selection, e.g. ``cyzone 15 4 -8 protein and
+    resid 42`` selects the center of geometry of protein and resid 42, and
+    creates a cylinder of external radius 15 centered on the COG. In z, the
+    cylinder extends from 4 above the COG to 8 below. Positive values for
+    *zMin*, or negative ones for *zMax*, are allowed.
+
+    .. versionchanged:: 0.10.0
+       keywords *cyzone* and *cylayer* now take *zMax* and *zMin* to be
+       relative to the COG of *selection*, instead of absolute z-values
+       in the box.
+
+point *x* *y* *z*  *distance*
+    selects all atoms within a cutoff of a point in space, make sure
+    coordinate is separated by spaces, e.g. ``point 5.0 5.0 5.0 3.5``
+    selects all atoms within 3.5 Angstroms of the coordinate (5.0, 5.0,
+    5.0)
+
+prop [abs] *property*  *operator*  *value*
+    selects atoms based on position, using *property* **x**, **y**, or
+    **z** coordinate. Supports the **abs** keyword (for absolute value) and
+    the following *operators*: **<, >, <=, >=, ==, !=**. For example,
+    ``prop z >= 5.0`` selects all atoms with z coordinate greater than 5.0;
+    ``prop abs z <= 5.0`` selects all atoms within -5.0 <= z <= 5.0.
+
+From version 0.6 onwards, some geometric selections (around, sphlayer,
```

Comment:
This section is outdated. Need to be clearer about periodicity handling – probably needs tobe linked to a whole page on this topic.

Reply:
I skimmed a few issues and some of the code. It looks like:

* a box is passed into every geometric keyword
* selection only uses the bruteforce or nsgrid methods for distances
* periodic=True by default if there's a box (and nsgrid creates a pseudo box if not, but brute force doesn't seem to?)
* because of the automatic method choosing, it's all reasonably fast
* and works for any triclinic box?